packages/server/lib/controllers/files.js

```diff
@@ -68,9 +68,12 @@ module.exports = {
       if (spec === '__all') {
         debug('returning all specs')
 
-        const pattern = await ctx.project.specPatternForTestingType(config.projectRoot, 'e2e')
+        const [e2ePatterns, componentPatterns] = await Promise.all([
+          ctx.project.specPatternsForTestingType(ctx.project.projectRoot, 'e2e'),
+          ctx.project.specPatternsForTestingType(ctx.project.projectRoot, 'component'),
+        ])
 
-        return ctx.project.findSpecs(config.projectRoot, 'e2e', pattern)
+        return ctx.project.findSpecs(config.projectRoot, 'e2e', e2ePatterns.specPattern, componentPatterns.specPattern)
```

Comment:
This is a bit unexpected...I assume we are adding the component specs to the ignore pattern incase the e2e pattern also matches it? Might be worth throwing a comment in here for clarity.

Reply:
Done, and yep this is confusing. I hope we can refactor & consolidate this soon, and change this file to be `.ts`. I'm trying to keep this PR to just renames, introducing at little actual logic as possible.

For interest, this was originally decided and documented here: https://www.notion.so/cypressdx/Changes-for-10-0-0-1dca213837b84d6ea297c2d12c615f3a#ac625f8f78fb474ca884ee64e7191e9b

Reply:
Done: https://github.com/cypress-io/cypress/pull/19345/commits/8ebfa3b5e819af282c7b639623f8c9b7e3c9d034